android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/service/adyen/AdyenRepository.java

```diff
@@ -0,0 +1,139 @@
+package com.appcoins.sdk.billing.service.adyen;
+
+import com.appcoins.sdk.billing.listeners.GetTransactionListener;
+import com.appcoins.sdk.billing.listeners.LoadPaymentInfoListener;
+import com.appcoins.sdk.billing.listeners.MakePaymentListener;
+import com.appcoins.sdk.billing.listeners.NoInfoResponseListener;
+import com.appcoins.sdk.billing.service.Service;
+import com.appcoins.sdk.billing.service.ServiceResponseListener;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AdyenRepository {
+
+  private Service bdsService;
+  private AdyenListenerProvider adyenListenerProvider;
+
+  public AdyenRepository(Service bdsService, AdyenListenerProvider adyenListenerProvider) {
+
+    this.bdsService = bdsService;
+    this.adyenListenerProvider = adyenListenerProvider;
+  }
+
+  public void loadPaymentInfo(String method, String value, String currency, String walletAddress,
+      final LoadPaymentInfoListener listener) {
+    Map<String, String> queries = new LinkedHashMap<>();
+    queries.put("wallet.address", walletAddress);
+    queries.put("price.value", value);
+    queries.put("price.currency", currency);
+    queries.put("method", method);
+    ServiceResponseListener serviceResponseListener =
+        adyenListenerProvider.createLoadPaymentInfoListener(listener);
+
+    bdsService.makeRequest("payment-methods", "GET", new ArrayList<String>(), queries, null,
+        serviceResponseListener);
+  }
+
+  public void makePayment(String adyenPaymentMethod, boolean shouldStorePaymentMethod,
+      String returnUrl, String value, String currency, String reference, String paymentType,
+      String walletAddress, String origin, String packageName, String metadata, String sku,
+      String callbackUrl, String transactionType, String developerWallet, String storeWallet,
+      String oemWallet, String userWallet, final MakePaymentListener listener) {
```

Comment:
Maybe create a AdyenParameters object to avoid sending this number of parameters